File: client/src/renderer/services/api/domains/health.ts
/**
 * Health API Client
 * Matches FastAPI health router: /health
 */

import { BaseApiClient } from '../base/apiClient';

// Types matching the FastAPI backend health responses
export interface BasicHealthResponse {
  status: string;
  timestamp: string;
  service: string;
}

export interface ServiceStatusDetail {
  status: string;
  message: string;
}

export interface DatabaseStatus {
  status: string;
  message: string;
}

export interface StorageStatus {
  status: string;
  message: string;
  details?: any;
}

export interface DetailedServiceStatus {
  overall_status: string;
  timestamp: string;
  services: {
    pdf_processing?: ServiceStatusDetail;
    email_ingestion?: ServiceStatusDetail;
    pdf_templates?: ServiceStatusDetail;
    container?: ServiceStatusDetail;
  };
  database: DatabaseStatus;
  storage: StorageStatus;
}

export interface ReadinessResponse {
  ready: boolean;
  message: string;
  timestamp: string;
}

/**
 * Health API client for system health monitoring
 * Endpoints: /health, /health/status, /health/ready
 */
export class HealthApiClient extends BaseApiClient {
  /**
   * Basic health check
   * GET /health/
   */
  async getHealth(): Promise<BasicHealthResponse> {
    return this.get<BasicHealthResponse>('/health/');
  }

  /**
   * Detailed service status check
   * GET /health/status
   */
  async getServiceStatus(): Promise<DetailedServiceStatus> {
    return this.get<DetailedServiceStatus>('/health/status');
  }

  /**
   * Readiness check for critical services
   * GET /health/ready
   */
  async getReadiness(): Promise<ReadinessResponse> {
    return this.get<ReadinessResponse>('/health/ready');
  }
}

File: eto_server/src/api/schemas/health.py
"""
Health API Schemas
Request/response models for health and status endpoints
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime
from api.schemas.common import APIResponse


class ServiceStatus(BaseModel):
    """Individual service status"""
    status: str = Field(..., description="Service status (healthy, unhealthy, error)")
    message: str = Field(..., description="Status description")
    details: Optional[Dict[str, Any]] = Field(default=None, description="Additional service details")


class WorkerStatus(BaseModel):
    """ETO worker status details"""
    enabled: bool
    running: bool
    paused: bool
    pending_runs: int
    processing_runs: int


class EtoServiceStatus(ServiceStatus):
    """ETO service status with worker information"""
    worker: Optional[WorkerStatus] = Field(default=None, description="ETO background worker status")


class DatabaseStatus(ServiceStatus):
    """Database connection status"""
    pass


class StorageStatus(ServiceStatus):
    """Storage system status"""
    pass


class HealthCheckResponse(APIResponse):
    """Basic health check response"""
    service: str = "ETO Server"
    status: str = "healthy"
    timestamp: datetime = Field(default_factory=datetime.utcnow)


class DetailedStatusData(BaseModel):
    """Detailed status data"""
    overall_status: str
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    services: Dict[str, ServiceStatus] = Field(default_factory=dict)
    database: DatabaseStatus
    storage: StorageStatus


class DetailedStatusResponse(APIResponse):
    """Detailed service status response"""
    data: Optional[DetailedStatusData] = Field(default=None)


class ReadinessCheckData(BaseModel):
    """Readiness check data"""
    ready: bool
    message: str
    timestamp: datetime = Field(default_factory=datetime.utcnow)